sphinx/ext/graphviz.py

```diff
@@ -243,31 +244,24 @@ def render_dot(self, code, options, format, prefix='graphviz'):
 
     if format == 'png':
         dot_args.extend(['-Tcmapx', '-o%s.map' % outfn])
+
     try:
-        p = Popen(dot_args, stdout=PIPE, stdin=PIPE, stderr=PIPE, cwd=cwd)
-    except FileNotFoundError:
+        ret = subprocess.run(dot_args, input=code.encode(), stdout=PIPE, stderr=PIPE,
+                             cwd=cwd, check=True)
+        if not path.isfile(outfn):
+            raise GraphvizError(__('dot did not produce an output file:\n[stderr]\n%s\n'
+                                   '[stdout]\n%s') % (ret.stderr, ret.stdout))
```

Comment:
`ret.stderr` and `ret.stdout` is not decoded. Using `%r` is better for this case.

Note: It is hard to determine the correct encodings. So it is better to show them as bytes.